### PR TITLE
Add example of Three.js model with shadow

### PR DESCRIPTION
**Pull request by birkskyum**
None of the 3D model examples we have contain a shadow. This example is a more advanced modification of simple Three.js model example, enabling a shadow, to demonstrate that we do support this and to give people a proper starting point for with shadows.

This has been requested quite a few times - the latest slack thread on the topic is [here](https://osmus.slack.com/archives/C03TFH5NE83/p1729649981995159).

Screenshot 2024-10-23 at 17 15 46
Screenshot 2024-10-23 at 17 17 35

### CODE REVIEW

**Comment by birkskyum** [comment]
> Nice! Thanks! I want to merge the PR related to elevation breaking change. BTW, can shadow be casted on a 3D terrain scene, or is it extremely complicated?

The shadow is just a three.js flat plane, so I imagine it could be troublesome with the terrain - it's not draping to the terrain in any way, but I haven't tried.

**Comment by birkskyum** [comment]
@HarelM , this example doesn't have an elevation profile, so I don't think it'll be affected by the breaking change on elevation.

**Comment by HarelM** [comment]
Right, good point!

**Review by HarelM — APPROVED** [review]

File: test/examples/add-3d-model-threejs-with-shadow.html [hunk 1]
```diff
@@ -0,0 +1,162 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <title>Add a 3D model with shadow with three.js</title>
```
Comment (HarelM) [inline test/examples/add-3d-model-threejs-with-shadow.html:5]:
```suggestion
    <title>Add a 3D model with shadow using three.js</title>
```

File: test/examples/add-3d-model-threejs-with-shadow.html [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    <meta property="og:description" content="Use a custom style layer with three.js to add a 3D model with Shadow to the map." />
```
Comment (HarelM) [inline test/examples/add-3d-model-threejs-with-shadow.html:6]:
```suggestion
    <meta property="og:description" content="Use a custom style layer with three.js to add a 3D model with shadow to the map." />
```

**Comment by birkskyum** [comment]
I'll make a new PR with the changes you point to here.

File: test/examples/add-3d-model-threejs-with-shadow.html [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        html,
+        body,
+        #map {
+            height: 100%;
+        }
+    </style>
+</head>
+
+<body>
+    <script src="https://unpkg.com/three@0.147.0/build/three.min.js"></script>
```
Comment (HarelM) [inline test/examples/add-3d-model-threejs-with-shadow.html:26]: Are we using latest version? Can we update the other examples to use the latest version?

Comment (birkskyum) [reply test/examples/add-3d-model-threejs-with-shadow.html:26]: I'll look into that, and try to update all the threejs examples.

File: test/examples/add-3d-model-threejs-with-shadow.html [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    <script src="https://unpkg.com/three@0.147.0/examples/js/loaders/GLTFLoader.js"></script>
+    <div id="map"></div>
+    <script>
+        const map = (window.map = new maplibregl.Map({
```
Comment (HarelM) [inline test/examples/add-3d-model-threejs-with-shadow.html:30]: Is seeing map to window mandatory?

**Review by HarelM — APPROVED** [review]
Added some minor nit-picking. Looks good otherwise.

**Comment by birkskyum** [comment]
Continued in 
- #4888

**Pull request merged**